### PR TITLE
Addition to massiv benchmarks, namely:

### PR DESCRIPTION
**Pull request by lehins**
* Pin newest version of massiv at master (0.3.5.0)
* Addition of parallel version of matrix multiplication
* Column slicing benchmark
* Fix for the lehins/massiv#78 slowdown

### CODE REVIEW

File: bench/Runtime.hs [hunk 1]
```diff
@@ -102,8 +102,8 @@ main = do
 
     --
       vMA = MA.fromList MA.Seq vList :: MA.Array MA.P MA.Ix1 Double
-      aMA = MA.resize' (MA.Ix2 n n) vMA :: MA.Array MA.P MA.Ix2 Double
-      bMA = MA.resize' (MA.Ix2 n n) $ MA.fromList MA.Seq uList :: MA.Array MA.P MA.Ix2 Double
+    aMA' <- MA.resizeM (MA.Sz2 n n) vMA :: IO (MA.Array MA.P MA.Ix2 Double)
```
Comment (Magalame) [inline bench/Runtime.hs]: Is there a specific reason for using the IO monad here instead of the `let` binding?

**Comment by Magalame** [comment]
Sorry for the delay, I'm not at home for a while. Good PR though. After running some more tests, the `env` thingy was definitely needed

File: bench/Runtime.hs (hunk 1 above)
Comment (lehins) [reply bench/Runtime.hs]: I just updated the PR switching back to `resize'`, but that is only to make benchmarks compatible with older version of `massiv-0.2.*`.

It is always better to use `resizeM` vs `resize'` (and all other functions that follow that patterns) cause then you fail earlier, if there is indeed an error. Consider this example:

```haskell
λ> a = (0 :. 0) ... (1 :. 5)
λ> a' = resize' (Sz1 6) a
λ> print (computeAs U a')
Array U *** Exception: SizeElementsMismatchException: (Sz (2 :. 6)) vs (Sz1 6)
λ> a'' <- resizeM (Sz1 6) a
*** Exception: SizeElementsMismatchException: (Sz (2 :. 6)) vs (Sz1 6)
```

**Comment by lehins** [comment]
@Magalame yeah, `env` is our friend :)

I just ran the benchmarks on both `massiv-0.2.8.0` vs `massiv-0.3.5.0` and numbers are nearly identical, so I think I can close lehins/massiv#78

I've introduced so many changes in the core functionality in massiv-0.3.* that I wasn't sure if there was any performance regressions. These benchmarks made me feel better about it, so thank you!

**Comment by Magalame** [comment]
I'm very glad they were helpful!

**Comment by Magalame** [comment]
@lehins I might have merged a tad too quickly, I see you've changed back to `resize'`

**Comment by lehins** [comment]
@Magalame It's all good. I changed it back to `resize'` so the bechmarks are backwards compatible with massiv-0.2.8

On a separate note, there is a new release of massiv-0.3.6 coming up which includes `identityMatrix` function if you wanna add it to the benchmarks too: https://github.com/lehins/massiv/blob/master/massiv/src/Data/Massiv/Array/Numeric.hs#L201

**Pull request merged**